util/sync_buckets.py

```diff
@@ -0,0 +1,207 @@
+        required=False,
+        nargs="+",
+        default=[],
+    )
+    parser.add_argument(
+        "--autodetect-regions",
+        action="store_true",
+        help="If set ec2.describe_regions is used to retrieve regions. "
+        "Additional regions (e.g. opt-in) can be specified with --regions",
+        required=False,
+        default=False,
+    )
+    parser.add_argument(
+        "--dest-bucket",
+        type=str,
+        help="Bucket to upload to, defaults to {region}-aws-parallelcluster",
+        required=False,
+        default="{region}-aws-parallelcluster",
+    )
+    parser.add_argument(
+        "--src-bucket", type=str, help="Source bucket", required=True,
+    )
+    parser.add_argument(
+        "--src-bucket-region", type=str, help="Source bucket region", required=True,
+    )
+    parser.add_argument("--src-files", help="Files to sync", nargs="+", required=True)
+    parser.add_argument(
+        "--credentials",
+        help="STS credential endpoint, in the format <region>,<endpoint>,<ARN>,<externalId>."
+        "Could be specified multiple times",
+        required=False,
+        nargs="+",
+        type=_aws_credentials_type,
+        default=[],
+    )
+    parser.add_argument(
+        "--deploy",
+        action="store_true",
+        help="If deploy is false, we will perform a dryrun and no file will be pushed to buckets",
+        default=False,
+        required=False,
+    )
+    parser.add_argument(
+        "--update-existing",
+        action="store_true",
+        help="Overrides existing files in dest buckets",
+        default=False,
+        required=False,
+    )
+
+    args = parser.parse_args()
+
+    if args.autodetect_regions:
+        args.regions.extend(get_aws_regions(args.partition))
+
+    _validate_args(args, parser)
+    return args
+
+
+def _get_s3_object_metadata(s3_url):
+    req = urllib.request.Request(url=s3_url, method="HEAD")
+    response = urllib.request.urlopen(req)
+    assert response.status == 200
+    metadata = {}
+    metadata["version_id"] = response.headers.get("x-amz-version-id")
+    metadata["size"] = response.headers.get("Content-Length")
+
+    return metadata
+
+
+def _md5_checksum(filename):
+    m = hashlib.md5()
+    with open(filename, "rb") as f:
+        for data in iter(lambda: f.read(1024 * 1024), b""):
+            m.update(data)
+    return base64.b64encode(m.digest()).decode("utf-8")
+
+
+def _upload_files(args, sts_credentials, dir):
+    for region in args.regions:
+        for file in args.src_files:
+            logging.info("Copying file %s to region %s", file, region)
+            doc_manager = S3DocumentManager(region, sts_credentials.get(region))
+            file_path = f"{dir}/{file}"
+            dest_bucket = args.dest_bucket.format(region=region)
+            exists = doc_manager.version_exists(s3_bucket=dest_bucket, document_s3_path=file)
+            if not args.update_existing and exists:
```

Comment:
I was obtaining the same check by unflagging the update-existing (in dryrun) and verifying that all files were already in destination, that is why I avoided adding an extra flag but I do agree that in general it might be useful. We could add it later on if we plan to use it.